### PR TITLE
[e2e] Add description of use case “storage version with non-existing id should be GC'ed” execution conditions

### PR DESCRIPTION
**Pull request by tanjing2020**
O# Please enter the commit message for your changes. Lines starting

#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Add description of use case “storage version with non-existing id should be GC'ed” execution conditions

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
@roycaihw 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by roycaihw** [comment]
/assign @caesarxuchao
for approval

File: test/e2e/apimachinery/storage_version.go [hunk 1]
```diff
@@ -34,6 +34,7 @@ const (
 	idNonExist = "id-non-exist"
 )
 
+// This test requires that --feature-gates=APIServerIdentity=true,StorageVersionAPI=true be set on the apiserver and the controller maneger
 var _ = SIGDescribe("StorageVersion resources [Feature:StorageVersionAPI]", func() {
```
Comment (caesarxuchao) [inline test/e2e/apimachinery/storage_version.go:38]: Can you add `[Feature:APIServerIdentity]` in the SIGDescribe?

File: test/e2e/apimachinery/storage_version.go [hunk 2]
```diff
@@ -34,6 +34,7 @@ const (
 	idNonExist = "id-non-exist"
 )
 
+// This test requires that --feature-gates=APIServerIdentity=true,StorageVersionAPI=true be set on the apiserver and the controller maneger
```
Comment (caesarxuchao) [inline test/e2e/apimachinery/storage_version.go:37]: nit: s/maneger/manager

File: test/e2e/apimachinery/storage_version.go [hunk 3]
```diff
@@ -34,7 +34,8 @@ const (
 	idNonExist = "id-non-exist"
 )
 
-var _ = SIGDescribe("StorageVersion resources [Feature:StorageVersionAPI]", func() {
+// This test requires that --feature-gates=APIServerIdentity=true,StorageVersionAPI=true be set on the apiserver and the controller manager
+var _ = SIGDescribe("StorageVersion resources [Feature:StorageVersionAPI, APIServerIdentity]", func() {
```
Comment (liggitt) [inline test/e2e/apimachinery/storage_version.go:38]: changing this tag will break the alpha e2e job config at https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes/sig-cloud-provider/gcp/gcp-gce.yaml#L413

**Pull request merged**